src/allmydata/web/status.py

```diff
@@ -35,187 +35,236 @@ def render_time(self, ctx, data):
-        return self._get_time("cumulative_sending")
+    @renderer
+    def time_cumulative_sending(self, req, tag):
+        return tag(self._get_time("cumulative_sending"))
 
-    def data_time_hashes_and_close(self, ctx, data):
-        return self._get_time("hashes_and_close")
+    @renderer
+    def time_hashes_and_close(self, req, tag):
+        return tag(self._get_time("hashes_and_close"))
 
     def _get_rate(self, name):
         d = self.upload_results()
         def _convert(r):
             file_size = r.get_file_size()
             duration = r.get_timings().get(name)
-            return compute_rate(file_size, duration)
+            return abbreviate_rate(compute_rate(file_size, duration))
         d.addCallback(_convert)
         return d
 
-    def data_rate_total(self, ctx, data):
-        return self._get_rate("total")
+    @renderer
+    def rate_total(self, req, tag):
+        return tag(self._get_rate("total"))
 
-    def data_rate_storage_index(self, ctx, data):
-        return self._get_rate("storage_index")
+    @renderer
+    def rate_storage_index(self, req, tag):
+        return tag(self._get_rate("storage_index"))
 
-    def data_rate_encode(self, ctx, data):
-        return self._get_rate("cumulative_encoding")
+    @renderer
+    def rate_encode(self, req, tag):
+        return tag(self._get_rate("cumulative_encoding"))
 
-    def data_rate_push(self, ctx, data):
+    @renderer
+    def rate_push(self, req, tag):
         return self._get_rate("cumulative_sending")
 
-    def data_rate_encode_and_push(self, ctx, data):
+    @renderer
+    def rate_encode_and_push(self, req, tag):
         d = self.upload_results()
         def _convert(r):
             file_size = r.get_file_size()
             time1 = r.get_timings().get("cumulative_encoding")
             time2 = r.get_timings().get("cumulative_sending")
             if (time1 is None or time2 is None):
-                return None
+                return abbreviate_rate(None)
             else:
-                return compute_rate(file_size, time1+time2)
+                return abbreviate_rate(compute_rate(file_size, time1+time2))
         d.addCallback(_convert)
         return d
 
-    def data_rate_ciphertext_fetch(self, ctx, data):
+    @renderer
+    def rate_ciphertext_fetch(self, req, tag):
         d = self.upload_results()
         def _convert(r):
             fetch_size = r.get_ciphertext_fetched()
             duration = r.get_timings().get("cumulative_fetch")
-            return compute_rate(fetch_size, duration)
+            return abbreviate_rate(compute_rate(fetch_size, duration))
         d.addCallback(_convert)
         return d
 
-class UploadStatusPage(UploadResultsRendererMixin, rend.Page):
-    docFactory = getxmlfile("upload-status.xhtml")
 
-    def __init__(self, data):
-        rend.Page.__init__(self, data)
-        self.upload_status = data
+class UploadStatusPage(MultiFormatResource):
+    """Renders /status/up-%d."""
```

Comment:
Is it really a `MultiFormatResource` or just a `Resource`?

Reply:
Made this change in 6ab7f26.